src/main/java/rx/exceptions/Exceptions.java

```diff
@@ -148,5 +166,25 @@ public static Throwable getFinalCause(Throwable e) {
         }
         return e;
     }
-
+    /**
+     * Throws a single or multiple exceptions contained in the collection, wrapping it into
+     * {@code CompositeException} if necessary.
+     * @param exceptions the collection of exceptions. If null or empty, no exception is thrown.
+     * If the collection contains a single exception, that exception is either thrown as-is or wrapped into a
+     * CompositeException. Multiple exceptions are wrapped into a CompositeException.
+     * @param whileText the circumstance string to be appended to the thrown CompositeException, inserted after
+     * the sentences "Exception" and "Multiple exceptions".
+     */
+    @Experimental
+    public static void throwIfAny(Collection<? extends Throwable> exceptions, String whileText) {
+        if (exceptions != null && !exceptions.isEmpty()) {
+            if (exceptions.size() == 1) {
+                Throwable t = exceptions.iterator().next();
+                throw propagate(t, "Exception" + whileText);
```

Comment:
This `throw` does nothing ... is this "phantom" behavior done so that the code compiles to be terminating here, even though it is inside the method that throws? If so that is tricksy!